src/main/java/org/mule/extension/email/internal/util/message/EmailMessage.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.email.internal.util.message;
+
+import static java.lang.String.format;
+import static org.mule.extension.email.internal.util.EmailUtils.getMultipart;
+
+import org.mule.extension.email.api.exception.EmailException;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import javax.mail.MessagingException;
+import javax.mail.Multipart;
+import javax.mail.Part;
+
+/**
+ * Abstraction to represent a email message, exposing its body text and attachments.
+ *
+ * @since 1.2.0
+ */
+public class EmailMessage {
+
+  private MessageBody body;
+
+  private Collection<MessageAttachment> attachments = new ArrayList<>();
+
+  public EmailMessage(Part message) {
+    try {
+      if (message.isMimeType("multipart/*")) {
+        Multipart mp = getMultipart(message);
+        Part first = mp.getBodyPart(0);
+        if (message.isMimeType("multipart/alternative") || first.isMimeType("multipart/alternative")) {
+          body = new AlternativeBody(message.isMimeType("multipart/alternative") ? message : first);
+        } else if (message.isMimeType("multipart/related") || first.isMimeType("multipart/related")) {
+          body = new SimpleBody(message.isMimeType("multipart/related") ? message : first);
+        } else if (message.isMimeType("multipart/mixed")) {
+          body = new SimpleBody(first);
+        } else {
+          throw new EmailException(format("Error processing part. Invalid/Unsupported MimeType: %s", message.getContentType()));
```

Comment:
What were we doing in this situation before?

Reply:
We were skipping I believe, I'll change the code to do the same then.